manifests/warranty.pp

```diff
@@ -5,12 +5,14 @@
 # Used by the fact in this module
 #
 class dell::warranty (
-  $api_key,
+  Optional[String] $api_key,
+  String $file_mode = '0644',
 ) {
   file { '/etc/dell_api_key':
-    content => $api_key,
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0644',
+    content   => $api_key,
+    owner     => 'root',
+    group     => 'root',
+    mode      => $file_mode,
+    show_diff => false,
```

Comment:
Instead of `show_diff => false`, we could use `Sensitive($api_key)` for the content.

Reply:
I believe the change I made addresses this.  I was going to use Sensitive to begin with because I thought this module still supported Puppet 4.  More reading of course told me it was also supported in Puppet 4, so that was my fault.  Thanks!